timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/AsyncLockService.java

```diff
@@ -142,6 +143,12 @@ public AsyncResult<Leased<LockToken>> lockImmutableTimestamp(UUID requestId, lon
         return immutableTimestampLockResult;
     }
 
+    public AsyncResult<Leased<LockToken>> acquireNamedTimestampLock(
```

Comment:
The implementations are in https://github.com/palantir/atlasdb/pull/7336 but did not want to couple the PRs yet